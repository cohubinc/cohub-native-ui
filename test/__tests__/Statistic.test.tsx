import React from "react";
import { mount } from "enzyme";

import { Statistic } from "dist";

it("renders without crashing", async () => {
  const wrapper = mount(
    <Statistic label="test stat" value={100} format="number" />
  );

  expect(wrapper).toMatchSnapshot();
});
