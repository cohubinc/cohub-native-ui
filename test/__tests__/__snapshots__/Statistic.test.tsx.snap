// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Statistic
  format="number"
  label="test stat"
  value={100}
>
  <div
    className="flex flex-column justify-center items-center"
  >
    <Component
      color="hsl(0, 2%, 12%)"
      weight="500"
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontWeight": "500",
            },
            Object {
              "color": "hsl(0, 2%, 12%)",
            },
            Object {
              "fontFamily": "Helvetica",
            },
          ]
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "color": "rgba(31,30,30,1.00)",
              "fontFamily": "Helvetica",
              "fontSize": "16px",
              "fontWeight": "500",
            }
          }
        >
          <FormatNumber
            value={100}
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowLeadingZeros={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator={true}
              thousandsGroupStyle="thousand"
              type="text"
              value={100}
            >
              <span>
                100
              </span>
            </NumberFormat>
          </FormatNumber>
        </div>
      </Text>
    </Component>
    <Component
      muted={true}
      uppercase={true}
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontWeight": "400",
            },
            Object {
              "color": "hsl(356, 3%, 59%)",
            },
            Object {
              "fontFamily": "Helvetica",
            },
            Object {
              "textTransform": "uppercase",
            },
          ]
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "color": "rgba(154,147,148,1.00)",
              "fontFamily": "Helvetica",
              "fontSize": "12px",
              "fontWeight": "400",
              "textTransform": "uppercase",
            }
          }
        >
          test stat
        </div>
      </Text>
    </Component>
  </div>
</Statistic>
`;
